### PR TITLE
[Linux] Fix playback of optical DVDs

### LINKED ISSUE #18896: DVD's not playing
### Describe the bug
Here is a clear and concise description of what the problem is:

Encrypted dvd's not playing on Kodi 19. They work under Kodi 18.

## Expected Behavior
Here is a clear and concise description of what was expected to happen:

Dvd's playing.

## Actual Behavior

Directly after starting dvd it jumps back to Kodi home menu.

## Possible Fix

Probably something to do with libdvdcss. When playing dvd in Kodi 19 it doesn't create folder under .dvdcss. However after playing dvd first on VLC and then going to Kodi (without ejecting dvd from drive) it plays fine.

### To Reproduce

1. Put encrypted dvd on drive
2. Try to play it on Kodi 19
3.

### Debuglog

The debuglog can be found here:
https://pastebin.ubuntu.com/p/Zm8MfH8pyj/

### Screenshots 
Here are some links or screenshots to help explain the problem:

## Additional context or screenshots (if appropriate)
Here is some additional context or explanation that might help:

Here is log from Kodi 18.9 after downgrading from Kodi 19 https://pastebin.ubuntu.com/p/Bwr3WJYfGy/
There's this line `DEBUG: libdvdread: Attempting to retrieve all CSS keys` what doesn't exists in log file from Kodi 19.

### Your Environment
Used Operating system:

 - [ ] Android
 - [ ] iOS
 - [ ] tvOS
 - [x] Linux
 - [ ] OSX
 - [ ] Windows
 - [ ] Windows UWP

 - Operating system version/name: Ubuntu 20.04
 - Kodi version: 19

*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

### PR DESCRIPTION
**Pull request by enen92**
## Description
This PR is an alternative to https://github.com/xbmc/xbmc/pull/19226 as I didn't like much the fact we're adapting the iso9660 when it wasn't actually used to process the DVD. This fixes the detection logic instead.
I didn't test blu-ray playback as I don't have any way to do it (@DaVukovic if possible) but it should work™

## Motivation and Context
Fixes https://github.com/xbmc/xbmc/issues/18896
Restores optical DVD playback in Kodi

## How Has This Been Tested?
Tested in linux and windows (testbuild: http://mirrors.kodi.tv/test-builds/windows/win64/KodiSetup-20210303-60a8f395-PR15-merge-x64.exe)

## Screenshots (if appropriate):
See https://github.com/xbmc/xbmc/pull/19226

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Review by howie-f — COMMENTED** [review]
this worsens with my test setup and disc. after hitting play i see a busy spinner forever and a high cpu load. only kill -9 stops it. discname detected and shown in the ui, as used to.

VLC installed afterwards still plays fine.

https://paste.kodi.tv/dumufufiji.kodi

Unbenannt

**Comment by enen92** [comment]
Hey @howie-f, this doesn't fix your previous issue it just makes the dvd detection work without trying to open the video.ifo using the iso9660:// vfs. The path check keeps there for legacy reasons only.
We have previously concluded that VLC always open the raw optical device (/dev/dvd is passed to libdvdnav/read) whereas Kodi sometimes pass the root mount point (/mnt/mydvd) and according to some comment in the code it should work too (probably by luck :)). For me it always plays regardless of the path. In your VirtualBox passthrough setup the second one doesn't play but I highly doubt that's something on the Kodi side.
You should try to investigate why /dev/cdrom or /dev/dvd is not returned as the optical path in your setup by libcdio. I can't help much there, but please do not block this PR under the assumption your setup is something common :)

**Comment by howie-f** [comment]
all good @enen92. that was just the user report, thought it may be of value. ofc not blocking, codewise i'm fine with this.

**Comment by DaVukovic** [comment]
For everything plays fine. CPU usage is at 20-24% with DVD playback and ~35% with BR-playback. So I can't complain in that regard. 

Tested with current Kodi master on Fedora and a 1050ti as a GPU

**Comment by enen92** [comment]
Thanks much for testing @DaVukovic. Glad to know BRs continue to play just fine

File: xbmc/storage/MediaManager.cpp [hunk 1]
```diff
@@ -716,29 +716,26 @@ CMediaManager::DiscInfo CMediaManager::GetDiscInfo(const std::string& mediaPath)
   if (mediaPath.empty())
     return info;
 
-  // Try finding VIDEO_TS/VIDEO_TS.IFO - this indicates a DVD disc is inserted
+  // Check for DVD discs
   std::string pathVideoTS = URIUtils::AddFileToFolder(mediaPath, "VIDEO_TS");
```
Comment (lrusak) [inline xbmc/storage/MediaManager.cpp:720]: This line isn't needed anymore as far as I can tell

File: xbmc/storage/MediaManager.cpp [hunk 2]
```diff
@@ -716,29 +716,26 @@ CMediaManager::DiscInfo CMediaManager::GetDiscInfo(const std::string& mediaPath)
...
-  if (CFile::Exists(URIUtils::AddFileToFolder(pathVideoTS, "VIDEO_TS.IFO")))
```
Comment (lrusak) [inline xbmc/storage/MediaManager.cpp:721]: would there ever be a case that a DVD wouldn't have this?

**Review by lrusak — CHANGES_REQUESTED** [review]
Looks alright. With the one requested change you can change some values from using `pathVideoTS` to `mediaPath` I believe.

File: xbmc/storage/MediaManager.cpp (hunk 1 above)
Comment (enen92) [reply xbmc/storage/MediaManager.cpp]: fixed

File: xbmc/storage/MediaManager.cpp (hunk 2 above)
Comment (enen92) [reply xbmc/storage/MediaManager.cpp:721]: There is at least a case where the video_ts.ifo is in the root folder and not under video_ts: https://github.com/xbmc/xbmc/blob/d6a4bc3f030d5e57e8e706ff5c759fa3e812eca0/xbmc/FileItem.cpp#L1829-L1832
Either way, this is used for dvd stacking / dvd folder storage on disk and not exactly to request playback from the optical drive (so a different code path). For this PR/fix if we manage to open the dvd with libdvdnav/read we can safely assume the item is a DVD.

**Comment by enen92** [comment]
jenkins build this please

**Comment by enen92** [comment]
Jenkins build this please

**Review by howie-f — APPROVED** [review]
DVD playback is stable with https://github.com/xbmc/xbmc/pull/19327/commits/dde4f4e7fc693cefd5752301a33776d9549c7e69 in my virtual box :+1:

**Review by lrusak — APPROVED** [review]
LGTM

**Comment by enen92** [comment]
jenkins build and merge

**Pull request merged**